dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -507,64 +507,58 @@ BlockInputStreamPtr Segment::getInputStreamForDataExport(const DMContext & dm_co
-///        del_mark != 0 filtering and sorted merge.
-///        We will just read all the data and return.
-///     2. We read in fast mode. (when filter_delete_mark = true)
-///        In this case, we will read all the data without mvcc filtering and sorted merge,
-///        but we will do del_mark != 0 filtering.
-BlockInputStreamPtr Segment::getInputStreamRaw(const DMContext & dm_context,
-                                               const ColumnDefines & columns_to_read,
-                                               const SegmentSnapshotPtr & segment_snap,
-                                               const RowKeyRanges & data_ranges,
-                                               const RSOperatorPtr & filter,
-                                               bool filter_delete_mark,
-                                               size_t expected_block_size)
+/// We call getInputStreamFast when we read in fast mode.
+/// In this case, we will read all the data in delta and stable, and then merge them without sorting.
+/// Besides, we will do del_mark != 0 filtering to drop the deleted rows.
+/// In conclusion, the output is unsorted, and does not do mvcc filtering.
+BlockInputStreamPtr Segment::getInputStreamFast(
+    const DMContext & dm_context,
+    const ColumnDefines & columns_to_read,
+    const SegmentSnapshotPtr & segment_snap,
+    const RowKeyRanges & data_ranges,
+    const RSOperatorPtr & filter,
+    size_t expected_block_size)
 {
-    /// Now, we use filter_delete_mark to determine whether it is in fast mode or just from `selraw * xxxx`
-    /// But this way seems not to be robustness enough, maybe we need another flag?
     auto new_columns_to_read = std::make_shared<ColumnDefines>();
 
     // new_columns_to_read need at most columns_to_read.size() + 2, due to may extra insert into the handle column and del_mark column.
     new_columns_to_read->reserve(columns_to_read.size() + 2);
 
     new_columns_to_read->push_back(getExtraHandleColumnDefine(is_common_handle));
-    if (filter_delete_mark)
-    {
-        new_columns_to_read->push_back(getTagColumnDefine());
-    }
-
-    bool enable_handle_clean_read = filter_delete_mark;
-    bool enable_del_clean_read = filter_delete_mark;
+    new_columns_to_read->push_back(getTagColumnDefine());
+
+    /// When we read in fast mode, we can try to do the following optimization:
+    /// 1. Handle Column Optimization:
+    ///     when the columns_to_read does not include HANDLE_COLUMN,
+    ///     we can try to skip reading the handle column if the pack's handle range is fully within read range.
+    ///     Thus, in this case, we set enable_handle_clean_read = true.
+    /// 2. Del Column Optimization:
+    ///    when the columns_to_read does not include TAG_COLUMN,
+    ///    we can try to skip reading the del column if the pack has no deleted rows.
+    ///    Thus, in this case, we set enable_del_clean_read = true.
+    /// 3. Version Column Optimization:
```

Comment:
In what cases will we need the version column in fast scan mode?

Reply:
We only need version column when the columns_to_read contains version_column. Otherwise, we will not read version column.

Reply:
Just curious, in what cases will the columns_to_read contains version_column? Will user be able to query for it?

Reply:
The user is not able to query it through the TiDB SQL client. Just keep for debugging when querying through ch-client